numba/cuda/tests/cudapy/test_array.py

```diff
@@ -61,5 +62,127 @@ def test_auto_device_const(self):
         d, _ = cuda.devicearray.auto_device(2)
         self.assertTrue(np.all(d.copy_to_host() == np.array(2)))
 
+    def _test_device_array_like_same(self, d_a):
```

Comment:
Was wondering in these new tests if there's merit in, where possible, using `np.empty()` then assigning a `np.arange()` into it and then checking that the data is correct as well as the strides/dtype/shape etc?

Reply:
I feel that to quite some extent this is already duplicated by either the `test_transpose_view` test (added below) or existing tests in `test_cuda_ndarray` and `test_cuda_array_slicing` etc. that test the functionality of device arrays, and that these new tests cover ensuring that `device_array_like` creates new devices arrays with the appropriate properties.

That said, I can add more tests if there are particular cases you have in mind - are there any that you'd really like to see here?

Reply:
I was just wondering about checking data out of habit, but your reasoning RE other tests is sound so I'm happy to leave as-is. Thanks.